start_coupled.R

```diff
@@ -283,22 +294,17 @@ start_coupled <- function(path_remind, path_magpie, cfg_rem, cfg_mag, runname, m
 
       if (all(gdx_exist | gdx_na)) {
         message("Starting subsequent run ", run)
-        # for the sbatch command set the number of tasks per node
-        if (subseq.env$cfg_rem$gms$optimization == "nash" && subseq.env$cfg_rem$gms$cm_nash_mode == "parallel") {
-          # for nash: set the number of CPUs per node to number of regions + 1
-          nr_of_regions <- length(unique(read.csv2(subseq.env$cfg_rem$regionmapping)$RegionCode)) + 1
-        } else {
-          # for negishi: use only one CPU
-          nr_of_regions <- 1
-        }
         logfile <- file.path("output", subseq.env$fullrunname, "log.txt")
         if (! file.exists(dirname(logfile))) dir.create(dirname(logfile))
         subsequentcommand <- paste0("sbatch --qos=", subseq.env$qos, " --job-name=", subseq.env$fullrunname, " --output=", logfile,
-        " --mail-type=END --comment=REMIND-MAgPIE --tasks-per-node=", nr_of_regions,
-        " --wrap=\"Rscript start_coupled.R coupled_config=", RData_file, "\"")
+        " --mail-type=END --comment=REMIND-MAgPIE --tasks-per-node=", subseq.env$nr_of_regions,
+        " ", subseq.env$sbatch, " --wrap=\"Rscript start_coupled.R coupled_config=", RData_file, "\"")
         message(subsequentcommand)
         if (length(needfulldatagdx) > 0) {
-          system(subsequentcommand)
+          exitCode <- system(subsequentcommand)
+          if (0 < exitCode) {
+            stop("sbatch command failed, check logs")
```

Comment:
What is the rational here to fail after the `sbatch` of a single subsequent run failed? Why shouldn't it try to start the other subsequent runs and do the post-processing? I would strongly advise to use an `errorsfound` approach here that only fails really at the end of the script.

Reply:
I think the rational is:

1. In production runs: if `sbatch` fails, slurm is broken, all bets are off, we might as well stop.
2. In test runs: if `sbatch --wait` fails, the actual run somehow broke, that means the test should fails as fast as possible.

Does that make sense?

Reply:
@1: I have no idea how this can happen, but I've heard of temporary cluster hiccups. So I would rather prefer to at least try to start the other runs, because everything that will be started somewhere in the night can save you hours.
@2: That makes sense. Can we use
```
if (0 < exitCode) {
    message("sbatch command failed, check logs")
    stopifnot(! grepl("--wait", subsequentcommand))
}
```
or something like that? And, by the way: which logs are you supposed to check, actually?

Reply:
all of the logs. Of all runs that were started. And the logs point nicely at each other, the output of the test points at the Base-rem-1 log, this log points at the NDC-rem-1 and Base-rem-2 log etc.

Reply:
> So I would rather prefer to at least try to start the other runs, because everything that will be started somewhere in the night can save you hours.

I'm not sure, it adds quite some complexity for something which I have never seen and nobody told me they have ever seen it (sbatch fails, but the *next* sbatch will not fail).

Reply:
Like, if we think this happens, I'd rather but in some logic which re-tries three times every `sbatch` call, it would actually solve things in this situation (sbatch fails, but the next sbatch will be fine)

Reply:
Ok, but even if `sbatch` fails, I would actually prefer to complete the runtime and compareScenario stuff that is below. Additionally, in my short history with REMIND and the cluster, I have seen many things nobody has seen before (thinking about `.lock`).

Reply:
And: If you use `qos=shorty` or something like that, `sbatch --qos=shorty --wrap="Rscript start.R --test"` fails without an actual problem with `sbatch`. So you can trigger that with a simple typo in your config.

Reply:
I guess if you have scenarios with different qos, then the script does something useful after failing to start another scenario with a different qos, yes. I'm not sure how useful given that it is missing some scenarios, but I guess you really rather debug slightly weird behaviour because one scenario is missing in the chain and re-start after fixing if that means the re-start is faster.